fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/api/event/player/BlockBreakEvents.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.player;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+public class BlockBreakEvents {
+	/**
+	 * Callback before a block is broken.
+	 * Only called on the server, however updates are synced with the client.
+	 *
+	 * <p>Upon return:
+	 * <ul><li>SUCCESS/PASS/CONSUME continues the default code for breaking the block
+	 * <li>FAIL cancels the block breaking action</ul>
+	 *
+	 * <p>Fields:
+	 * <ul><li> world - The world at which the block is being broken
+	 * <li> player - The player who is breaking the block
+	 * <li> pos - The position at which the block is being broken
+	 * <li> state - The block state from BEFORE the block is broken
+	 * <li> entity - The block entity from BEFORE the block is broken (can be null)
+	 * <li> block - The block instance of the block that is being broken</ul>
```

Comment:
Document on the method in the functional interface, not here. This is an absolute mess

Reply:
I don't normally document my code, I will move it there

Reply:
I actually think this is the right place.
See the lifecycle events for example: https://github.com/FabricMC/fabric/blob/1.16/fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/api/client/event/lifecycle/v1/ClientChunkEvents.java